apps/speed.c

```diff
@@ -561,52 +615,517 @@ static OPT_PAIR ecdh_choices[] = {
 };
 #endif
 
+#ifndef SIGALRM
+# define COND(d) (count < (d))
+# define COUNT(d) (d)
+#else
+# define COND(c) (run && count<0x7fffffff)
+# define COUNT(d) (count)
+# ifndef _WIN32
+/*    signal(SIGALRM, sig_done); */
+# endif
+#endif                         /* SIGALRM */
```

Comment:
My understanding is that the purpose of this code is to define the condition (`COND`) used in the for loops depending on the availability of the function `signal()`

I moved it here because I will use `COND` in the all the `*_loop()` functions, but I don't actually need to call `signal()` here, which is why I have commented it out.

I think this code could be simplified to remove the `_WIN32` case entirely:
```c
#ifndef SIGALRM
# define COND(d) (count < (d))
# define COUNT(d) (d)
#else
# define COND(c) (run && count<0x7fffffff)
# define COUNT(d) (count)
#endif
```

With regard to the second definition (around line 1760) we don't need to redefine `COND` again, but we need to call `signal(SIGALRM, sig_done)`

The code would then become:

```c
/* not worth fixing */
#  error "You cannot disable DES on systems without SIGALRM."
# endif                        /* OPENSSL_NO_DES */
#else
# ifndef _WIN32
    signal(SIGALRM, sig_done);
# endif
#endif                         /* SIGALRM */
```